web/src/message_edit.js

```diff
@@ -1297,7 +1297,7 @@ export function move_topic_containing_message_to_stream(
     });
 }
 
-export function with_first_message_id(stream_id, topic_name, success_cb, error_cb) {
+export function with_last_message_id(stream_id, topic_name, success_cb, error_cb) {
```

Comment:
Please rename this in its own commit, and I think we should call this `with_newest_message_id` for clarity, and its cousin `with_oldest_message_id`?

Reply:
To avoid multiple calls to the server, the new code defines `is_message_oldest_or_newest` function which checks for the presence of the next/prev message in the topic by using the near narrow through a single server call.

Reply:
This removes the need for the `with_last_message_id` function. The `with_first_message_id` function is still intact and being used at other places in the code. If it is still relevant I can add the commit for just renaming `with_first_message_id` to `with_newest_message_id`.